src/commonMain/kotlin/com/charleskorn/kaml/YamlNamingStrategy.kt

```diff
@@ -0,0 +1,97 @@
+/*
+
+   Copyright 2018-2021 Charles Korn.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+package com.charleskorn.kaml
+
+public fun interface YamlNamingStrategy {
+    public fun serialNameForYaml(serialName: String): String
+
+    public companion object Builtins {
```

Comment:
Would be good to add some tests specifically for these implementations (eg. test that they behave correctly with empty strings, single characters, single words, multiple words etc.)

Reply:
A blank serial name isn't possible, unless `@SerialName` is used. In this case, the result is:
```kotlin
@Serializable
data class EmptySerialName(@SerialName("") val name: String)
```
Expected (or perhaps an exception):
```yaml
": "value""
```
Actual:
```yaml
"?
: "value""
```

This happens even in the absence of any naming strategy as blank serial names are not expected or intended. If it's not intentional, it's out of the scope of this PR to validate that behaviour.